libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -524,11 +532,80 @@ class MapboxNavigationTest {
 
         createMapboxNavigation()
 
-        mapboxNavigation.postUserFeedback(mockk(), mockk(), mockk(), mockk(), mockk())
-        mapboxNavigation.postUserFeedback(mockk(), mockk(), mockk(), mockk(), mockk(), mockk())
+        mapboxNavigation.postUserFeedback(
+            FEEDBACK_TYPE, DESCRIPTION, FEEDBACK_SOURCE, SCREENSHOT,
+            FEEDBACK_SUBTYPE, mockk<UserFeedbackObserver>(),
+        )
+        mapboxNavigation.postUserFeedback(
+            FEEDBACK_TYPE, DESCRIPTION, FEEDBACK_SOURCE,
+            SCREENSHOT, FEEDBACK_SUBTYPE, mockk(), mockk(),
+        )
 
         verify(exactly = 0) {
-            MapboxNavigationTelemetry.postUserFeedback(any(), any(), any(), any(), any(), any())
+            MapboxNavigationTelemetry.postUserFeedback(
+                any(), any(), any(), any(), any(), any(), any(),
+            )
+        }
+    }
+
+    @Test
+    fun `postUserFeedback sends feedback without metadata`() {
+        createMapboxNavigation()
+        val userFeedbackObserver = mockk<UserFeedbackObserver>()
+
+        mapboxNavigation.postUserFeedback(
+            FEEDBACK_TYPE, DESCRIPTION, FEEDBACK_SOURCE, SCREENSHOT,
+            FEEDBACK_SUBTYPE, userFeedbackObserver,
+        )
+
+        verify(exactly = 1) {
+            MapboxNavigationTelemetry.postUserFeedback(
+                FEEDBACK_TYPE, DESCRIPTION, FEEDBACK_SOURCE, SCREENSHOT,
+                FEEDBACK_SUBTYPE, feedbackMetadata = null, userFeedbackObserver,
+            )
+        }
+    }
+
+    @Test
+    fun `postUserFeedback sends feedback with metadata`() {
+        createMapboxNavigation()
+        val feedbackMetadata = mockk<FeedbackMetadata>()
+        val userFeedbackObserver = mockk<UserFeedbackObserver>()
+
+        mapboxNavigation.postUserFeedback(
+            FEEDBACK_TYPE, DESCRIPTION, FEEDBACK_SOURCE, SCREENSHOT,
+            FEEDBACK_SUBTYPE, feedbackMetadata, userFeedbackObserver,
+        )
+
+        verify(exactly = 1) {
+            MapboxNavigationTelemetry.postUserFeedback(
+                FEEDBACK_TYPE, DESCRIPTION, FEEDBACK_SOURCE, SCREENSHOT,
+                FEEDBACK_SUBTYPE, feedbackMetadata, userFeedbackObserver,
+            )
+        }
+    }
+
+    @Test
+    fun `register user feedback observer`() {
+        createMapboxNavigation()
+        val userFeedbackObserver = mockk<UserFeedbackObserver>()
+
+        mapboxNavigation.registerUserFeedbackObserver(userFeedbackObserver)
+
+        verify(exactly = 1) {
+            MapboxNavigationTelemetry.registerUserFeedbackObserver(userFeedbackObserver)
+        }
+    }
+
+    @Test
+    fun `unregister user feedback observer`() {
+        createMapboxNavigation()
+        val userFeedbackObserver = mockk<UserFeedbackObserver>()
+
+        mapboxNavigation.unregisterUserFeedbackObserver(userFeedbackObserver)
+
+        verify(exactly = 1) {
+            MapboxNavigationTelemetry.unregisterUserFeedbackObserver(userFeedbackObserver)
```

Comment:
Just want to share my opinion. You can ignore it and merge the PR if you don't agree or have a bad mood to continue discussion. The advice won't fix something critically important, but I believe it may improve your development experience 🙂 

This two tests test a very straight forward implementation detail. Have you ever had a case in your experience when those types of tests found a regression? By "found a regression" I mean a case when you had changed implementation details and thought that everything's fine, but a test told you that feature has a bug? 
From my experience it never happens when you test implementation. What usually happens with implementation tests is that they fail when you change implementation details and you have to rewrite them. The only benefit I see from implementation testing is increasing code coverage.

Instead I suggest you test a scenario which is similar to a real use case:
```kotlin
val testMapboxTelemetry = mockk<MapboxTelemetry>() // create a test version of mapbox telemetry with a behaviour which is similar to real.
// For example when it should call an observer when somebody post a feedback event
// my usual practice is write test double once and reuse it in different tests. 
createMapboxNavigation()


val testObserver = UserFeedbackEventRecoder() // you can use mockk + verify instead
mapboxNavigation.registerUserFeedbackObserver(testObserver)
mapboxNavigation.postUserFeedback(....)
mapboxNavigation.unregisterUserFeedbackObserver(testObserver)
mapboxNavigation.postUserFeedback(....)

testOserver.verifyEventsCount(1)
val event = testObserver.getEvent(1)
// compare fields in event with passed data
```

The suggested test doesn't test implementation details. It verifies a behaviour of system under test with a test double which has similar behaviour to a real object. Sometimes it's even easier to use a real object instead of test double, but I'm not sure that it's the case for mapbox telemetry. You won't need to rewrite this test if you change implementation detail.